app/controllers/workflow_service_controller.rb

```diff
@@ -0,0 +1,80 @@
+class WorkflowServiceController < ApplicationController
+  ##
+  # Is a document closeable?
+  def closeable
+    @status = check_if_can_close_version
+    render json: @status
+  end
+
+  ##
+  # Is a document openable?
+  def openable
+    @status = check_if_can_open_verison
+    render json: @status
+  end
+
+  ##
+  # Has an object been published?
+  def published
+    @status = check_if_published
+    render json: @status
+  end
+
+  ##
+  # Has an object been submitted?
+  def submitted
+    @status = check_if_submitted
+    render json: @status
+  end
+
+  ##
+  # Has an object been accessioned?
+  def accessioned
+    @status = check_if_accessioned
+    render json: @status
+  end
+
+  private
+
+  ##
+  # Ported over logic from app/helpers/dor_object_helper.rb#LN119
+  # @return [Boolean]
+  def check_if_published
+    return true if Dor::WorkflowService.get_lifecycle('dor', params[:pid], 'published')
```

Comment:
it would be nice to start having the `Dor::WorkflowService` calls put into a helper. Something like this perhaps?

```ruby
# untested example
def lifecycle_is?(druid, status, active_only = false)
  fn = (active_only ? :get_active_lifecycle : :get_lifecycle)
  Dor::WorkflowService.send(fn, 'dor', druid, status).is_a?(Time) ? true : false
end
```

over time it would help us isolate which workflow service calls are actually being made by Argo vs. dor-services, etc. maybe that's outside the scope of this PR...